### PR TITLE
ubuntu_pipeline: Added sign-off stage

### PR DESCRIPTION
**Pull request by adriansuhov**
The pipeline will now wait for 3 days for the owner to sign off the kernel. Different mails will be sent accordingly.

### CODE REVIEW

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation [hunk 1]
```diff
@@ -238,28 +238,56 @@ if (env.performance == "yes") {
...
 if (env.sendMail == "yes"){
     node ("meta_slave") {
+        echo "Send email with kernel validation results link"
+        if (env.kernel == "linux-azure") {
+            version_identifier = "_azure"
+        } else {
+            version_identifier = "_edge"
+        }
+        kernels_info = readFile (LATEST_VERSION_LOCATION)
+        kernels_info = kernels_info.split("\n")
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: make sure that the kernels_info is not null or empty, to not send useless information.

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation [hunk 2]
```diff
@@ -238,28 +238,56 @@ if (env.performance == "yes") {
...
+        stage('Kernel-SignOff') {
+            withCredentials([string(credentialsId: 'MAIL_LIST_SIGNOFF', variable: 'MAIL_LIST_SIGNOFF')]) {
+                try {
+                    checkout scm
+                    echo "Ask for manual sign-off from the owner"
+                    timeout (time: 72, unit: 'HOURS') {
+                        input(message: "Are you sure you want to sign off ${env.distro} ${latest_kernel_version} ${env.kernel} kernel?")
+                    }
+                    emailext (
+                        subject: "${env.distro} ${latest_kernel_version} ${env.kernel} is OK and can be signed off!",
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: remove the ! from the end

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation [hunk 3]
```diff
@@ -238,28 +238,56 @@ if (env.performance == "yes") {
...
+                        to:"${env.MAIL_LIST_SIGNOFF}",
+                        mimeType : "text/html",
+                        body: """
+                            Hello,<br/><br/>
+                            ${latest_kernel_version} ${env.kernel} kernel for ${env.distro} is OK and can be signed off!<br/><br/>
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: ditto.

Also send who did press the yes button, you can retrieve that from the input stage (the email of the one who pressed on sign off button).

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation [hunk 4]
```diff
@@ -238,28 +238,56 @@ if (env.performance == "yes") {
...
+                            Thank you,<br/>Jenkins CI
+                        """
+                    )
+                } catch (exc) {
+                    emailext (
+                        subject: "${latest_kernel_version} ${env.kernel} kernel for ${env.distro} can't be signed off!",
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: ditto ! mark

ditto who did not sign off,

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation (hunk 1 above)
Comment (adriansuhov) [reply linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: will change

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation (hunk 2 above)
Comment (adriansuhov) [reply linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: will change

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation (hunk 4 above)
Comment (adriansuhov) [reply linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: will change

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation (hunk 3 above)
Comment (adriansuhov) [reply linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: will change

**Comment by adriansuhov** [comment]
Changes have been made

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation [hunk 5]
```diff
@@ -238,28 +238,66 @@ if (env.performance == "yes") {
...
-                    subject: "Test results for ${env.distro} ${latest_kernel_version} ${env.kernel} kernel",
-                    to: "${env.MAIL_LIST}",
+                    subject: "Please sign off ${latest_kernel_version} ${env.kernel} kernel for ${env.distro}",
+                    to: "${env.MAIL_LIST_OWNER}",
                     mimeType : "text/html",
                     body: '${SCRIPT, template="ubuntu.template"}'
                 )
             }
         }
+        stage('Kernel-SignOff') {
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: use camel_case for stage names, as you are already using that convention before

Comment (ader1990) [reply linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: or Camel-Case, as I see you are using that for validation stages

**Pull request merged**